### ISSUE TITLE
build_rawframes.py --resume doesn't work correctly

### ISSUE THREAD

**Issue by Thinksky5124** [open-mmlab/mmaction2#1036]
When use extrcact_frames.sh ,the code is running in build_rawframes.py files. But the args "--resume" doesn't work correctly.

```python
if args.input_frames:
        print('Reading rgb frames from folder: ', args.src_dir)
        fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
        done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of rgb frame folders found: ', len(fullpath_list))
    else:
        print('Reading videos from folder: ', args.src_dir)
        if args.mixed_ext:
            print('Extension of videos is mixed')
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        else:
            print('Extension of videos: ', args.ext)
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level + '.' +
                                      args.ext)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of videos found: ', len(fullpath_list))

if args.resume:
    fullpath_list = set(fullpath_list).difference(set(done_fullpath_list))
    fullpath_list = list(fullpath_list)
    print('Resuming. number of videos to be done: ', len(fullpath_list))
```

I think it should modify by follow code:

```python
if args.input_frames:
        print('Reading rgb frames from folder: ', args.src_dir)
        fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
        done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of rgb frame folders found: ', len(fullpath_list))
    else:
        print('Reading videos from folder: ', args.src_dir)
        if args.mixed_ext:
            print('Extension of videos is mixed')
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        else:
            print('Extension of videos: ', args.ext)
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level + '.' +
                                      args.ext)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of videos found: ', len(fullpath_list))

if args.resume:
    fullpath_video_name_list = []
    done_video_name_list = []
    for fullpath in fullpath_list:
        fullpath_video_name_list.append(fullpath.split('/')[6].split('.')[0])
    for done_fullpath in done_fullpath_list:
        done_video_name_list.append(done_fullpath.split('/')[6].split('.')[0])
    fullpath_list = set(fullpath_video_name_list).difference(set(done_video_name_list))
    fullpath_list = list(fullpath_list)
    for i in range(len(fullpath_list)):
        fullpath_list[i] = '../../data/thumos14/videos/test/' + fullpath_list[i] + '.mp4'
      
      print('Resuming. number of videos to be done: ', len(fullpath_list))
```

this row:
`fullpath_list[i] = '../../data/thumos14/videos/test/' + fullpath_list[i] + '.mp4'`
could modify more but in my computer it works.

**Comment by dreamerlin** [comment]
could you please give more information? It seems the original codes are correct and the provided ones are not general

**Comment by Thinksky5124** [comment]
the file tree is
```
mmaction2
├── mmaction
├── tools
├── configs
├── data
│   ├── thumos14
│   │   ├── proposals
│   │   |   ├── thumos14_tag_val_normalized_proposal_list.txt
│   │   |   ├── thumos14_tag_test_normalized_proposal_list.txt
│   │   ├── annotations_val
│   │   ├── annotations_test
│   │   ├── videos
│   │   │   ├── val
│   │   │   |   ├── video_validation_0000001.mp4
│   │   │   |   ├── ...
│   │   |   ├── test
│   │   │   |   ├── video_test_0000001.mp4
│   │   │   |   ├── ...
│   │   ├── rawframes
│   │   │   ├── val
│   │   │   |   ├── video_validation_0000001
|   │   │   |   │   ├── img_00001.jpg
|   │   │   |   │   ├── img_00002.jpg
|   │   │   |   │   ├── ...
|   │   │   |   │   ├── flow_x_00001.jpg
|   │   │   |   │   ├── flow_x_00002.jpg
|   │   │   |   │   ├── ...
|   │   │   |   │   ├── flow_y_00001.jpg
|   │   │   |   │   ├── flow_y_00002.jpg
|   │   │   |   │   ├── ...
│   │   │   |   ├── ...
│   │   |   ├── test
│   │   │   |   ├── video_test_0000001
```
I have extracted all video frames.
If I run this command
`python build_rawframes.py ../../data/thumos14/videos/val/ ../../data/thumos14/rawframes/val/ --num-worker 40 --level 1 --flow-type tvl1 --ext mp4 --task both --resume`

with code 
```
if args.input_frames:
        print('Reading rgb frames from folder: ', args.src_dir)
        fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
        done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of rgb frame folders found: ', len(fullpath_list))
    else:
        print('Reading videos from folder: ', args.src_dir)
        if args.mixed_ext:
            print('Extension of videos is mixed')
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        else:
            print('Extension of videos: ', args.ext)
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level + '.' +
                                      args.ext)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of videos found: ', len(fullpath_list))

if args.resume:
    fullpath_list = set(fullpath_list).difference(set(done_fullpath_list))
    fullpath_list = list(fullpath_list)
    print('Resuming. number of videos to be done: ', len(fullpath_list))
```
it should not over write the rawflames that has been extract , but get the fullpath_list follow:
```
fullpath_list[0]:'../../data/thumos14/videos/val/video_validation_0000254.mp4'
fullpath_list[1]:'../../data/thumos14/videos/val/video_validation_0000390.mp4'
fullpath_list[2]:'../../data/thumos14/videos/val/video_validation_0000573.mp4'
.....
```
and print this output:
```
Reading videos from folder:  data/thumos14/videos/val
Extension of videos:  mp4
Total number of videos found:  1010
Resuming. number of videos to be done:  1010
```
It will overwrite all raw frames that has been extracted.

I think it is out of work , because of follow code:
```
fullpath_list = set(fullpath_list).difference(set(done_fullpath_list))
```
fullpath_list is:
```
fullpath_list[0]:'../../data/thumos14/videos/val/video_validation_0000254.mp4'
fullpath_list[1]:'../../data/thumos14/videos/val/video_validation_0000390.mp4'
fullpath_list[2]:'../../data/thumos14/videos/val/video_validation_0000573.mp4'
.....
```
done_fullpath_list is:
```
done_fullpath_list[0]:'../../data/thumos14/rawframes/val/video_validation_0000959'
done_fullpath_list[1]:'../../data/thumos14/rawframes/val/video_validation_0000278'
done_fullpath_list[2]:'../../data/thumos14/rawframes/val/video_validation_0000703'
....
```
use `set.difference(...)` will not select out videos that haven't been extract frames.

But if use my code 
```
if args.input_frames:
        print('Reading rgb frames from folder: ', args.src_dir)
        fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
        done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of rgb frame folders found: ', len(fullpath_list))
    else:
        print('Reading videos from folder: ', args.src_dir)
        if args.mixed_ext:
            print('Extension of videos is mixed')
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        else:
            print('Extension of videos: ', args.ext)
            fullpath_list = glob.glob(args.src_dir + '/*' * args.level + '.' +
                                      args.ext)
            done_fullpath_list = glob.glob(args.out_dir + '/*' * args.level)
        print('Total number of videos found: ', len(fullpath_list))

if args.resume:
    fullpath_video_name_list = []
    done_video_name_list = []
    for fullpath in fullpath_list:
        fullpath_video_name_list.append(fullpath.split('/')[6].split('.')[0])
    for done_fullpath in done_fullpath_list:
        done_video_name_list.append(done_fullpath.split('/')[6].split('.')[0])
    fullpath_list = set(fullpath_video_name_list).difference(set(done_video_name_list))
    fullpath_list = list(fullpath_list)
    for i in range(len(fullpath_list)):
        fullpath_list[i] = '../../data/thumos14/videos/val/' + fullpath_list[i] + '.mp4'
      
      print('Resuming. number of videos to be done: ', len(fullpath_list))
```
fullpath_video_name_list is:
```
fullpath_video_name_list[0]:'video_validation_0000923'
fullpath_video_name_list[1]:'video_validation_0000772'
fullpath_video_name_list[2]:'video_validation_0000895'
.....
```
done_video_name_list is:
```
done_video_name_list[0]:'video_validation_0000959'
done_video_name_list[1]:'video_validation_0000278'
done_video_name_list[2]:'video_validation_0000703'
....
```
it will correctly output :
```
Reading videos from folder:  data/thumos14/videos/val
Extension of videos:  mp4
Total number of videos found:  1010
Resuming. number of videos to be done:  0
```

**Comment by dreamerlin** [comment]
Sorry for the late reply. It will be fixed it in #1150
